tests/codex/testvalidation.nim

```diff
@@ -41,12 +56,170 @@ asyncchecksuite "validation":
   test "the list of slots that it's monitoring is empty initially":
     check validation.slots.len == 0
 
+  for (validationGroups, groupIndex) in [(100, 100'u16), (100, 101'u16)]:
+    test "initializing ValidationConfig fails when groupIndex is " &
+        "greater than or equal to validationGroups " &
+        fmt"(testing for {groupIndex = }, {validationGroups = })":
+      let groups = ValidationGroups(validationGroups).some
+      let validationConfig = ValidationConfig.init(
+          maxSlots, groups = groups, groupIndex = groupIndex)
+      check validationConfig.isFailure == true
+      check validationConfig.error.msg == "The value of the group index " &
+          "must be less than validation groups! " &
+          fmt"(got: {groupIndex = }, groups = {!groups})"
+  
+  test "initializing ValidationConfig fails when maxSlots is negative":
+    let maxSlots = -1
+    let validationConfig = ValidationConfig.init(
+        maxSlots = maxSlots, groups = ValidationGroups.none)
+    check validationConfig.isFailure == true
+    check validationConfig.error.msg == "The value of maxSlots must " &
+        fmt"be greater than or equal to 0! (got: {maxSlots})"
+  
+  test "initializing ValidationConfig fails when maxSlots is negative " &
+      "(validationGroups set)":
+    let maxSlots = -1
+    let validationConfig = ValidationConfig.init(
+        maxSlots = maxSlots, groups = validationGroups, groupIndex)
+    check validationConfig.isFailure == true
+    check validationConfig.error.msg == "The value of maxSlots must " &
+        fmt"be greater than or equal to 0! (got: {maxSlots})"
+
+  test "group index is irrelevant if validation groups are not set":
+    randomize()
+    for _ in 0..<100:
+      let groupIndex = rand(1000).uint16
```

Comment:
🟡 If we want to fuzz the input value, then perhaps we should cover the entire `uint16` space, eg use `rand(uint16.high.int).uint16` (bit ugly). 

Perhaps a more efficient test would be to test the boundaries (upper and lower) and then test a valid `uint16` number. Even that is a bit much honestly. Simply testing a valid `uint16` is probably enough since there are no "disallowed" exceptions.

Reply:
Using the "fuzz" is intended to communicate to the reader that the value really does not matter here. It is more to communicate to the reader the intended behaviour. I like more the option of covering the whole space, even though `rand(uint16.high.int).uint16` is a bit noisy. It is more what I intended.

Reply:
The compiler will only let us use valid `uint16` value in the `init` function, and there are no exceptions in the logic or logic branching that would result in a failed `init` function for specific `uint16` values. So I'm failing to understand how this is better than choosing a single valid `uint16` value to test.

Reply:
I am ok with using one value. I will change that.